### ISSUE TITLE
Getting an error saying : Expected a JavaScript module script but the server responded with a MIME type of "text/html" 

### ISSUE THREAD

**Issue by litcodeur** [mathe42/vite-plugin-comlink#136]
When I'm trying to use this plugin to work with webworkers, I'm getting the following error message in my browser console.

```
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
```

<img width="1211" alt="Screenshot 2024-04-27 at 4 16 46 PM" src="https://github.com/mathe42/vite-plugin-comlink/assets/52537686/992de0ed-f56a-4986-aac5-7d704733251c">

And when I look into the path that it tries to fetch manually by opening it in a new tab, it returns a not found but the type of content returned by server is of html.

Lmk if there's more info I can provide to help out with the issue,
Thanks!

**Comment by litcodeur** [comment]
I think, this might be a bug, so when named my worker file `foo.worker.ts`, it failed with the above error message, but when I renamed it to `fooworker.ts` (without a dot separator) it worked as expected.

**Comment by mathe42** [comment]
Please provide Code for a reproduction. And what Browser etc. you use. 

Does this only happen in dev Mode or also production?

Also this might be an upstream issue in vite.

**Comment by litcodeur** [comment]
Here's a codesandbox with a minimal reproduction: https://codesandbox.io/p/devbox/charming-albattani-ktmypy 

I'm using Arc browser (chromium based), on Mac device (M series). 

Haven't built and pushed changes to verify, but can confirm happens in dev mode.

**Comment by mathe42** [comment]
Workaround:

```ts
new URL("./worker.fail.ts", import.meta.url)
```

include extension in url.

I'm currently evaluating if this is a problem with this plugin or upstream in vite.

**Comment by mathe42** [comment]
It is a upstream Problem as 
```ts
const instance = new Worker(
              new URL("./worker.fail", import.meta.url),
              {
                /* normal Worker options*/
              },
            );
```

Also fails.

**Comment by mathe42** [comment]
fixed with v5.0.0

**Comment by litcodeur** [comment]
Thanks!
